fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1457,6 +1457,7 @@ private void executeAnalyze(AnalyzeStmt analyzeStmt, AnalyzeStatus analyzeStatus
         statsConnectCtx.getSessionVariable().setStatisticCollectParallelism(
                 context.getSessionVariable().getStatisticCollectParallelism());
         statsConnectCtx.setThreadLocalInfo();
+        statsConnectCtx.setStatisticsConnection(true);
```

Comment:
it would be set in the `StatisticExecutor`, is it necessary to set it here ?

Reply:
not all statistic SQL need set the variable, only collect job need